test/yugabyte_test.go

```diff
@@ -0,0 +1,282 @@
+	commandConnectYSQLSH := fmt.Sprintf("cd /home/%s/yugabyte-db/tserver && ./bin/ysqlsh --echo-queries -h %s -c \\\\conninfo", sshUser, privateIP)
+	retry.DoWithRetry(t, "Conecting YSQLSH", maxRetries, timeBetweenRetries, func() (string, error) {
+		output, err := ssh.CheckSshCommandE(t, sshSession, commandConnectYSQLSH)
+		if err != nil {
+			return "", err
+		}
+		logger.Logf(t, "Output of Ysql command :- %s", output)
+		return "", nil
+	})
+}
+
+// Verify the YCQL API
+func testYugaByteCQLSH(t *testing.T, terraformOptions *terraform.Options, privateIP string, sshUser string, sshSession ssh.Host) {
+	commandConnectCQLSH := fmt.Sprintf("cd /home/%s/yugabyte-db/tserver && ./bin/ycqlsh %s 9042 --execute 'SHOW HOST'", sshUser, privateIP)
+	retry.DoWithRetry(t, "Conecting YCQLSH", maxRetries, timeBetweenRetries, func() (string, error) {
+		output, err := ssh.CheckSshCommandE(t, sshSession, commandConnectCQLSH)
+		if err != nil {
+			return "", err
+		}
+		logger.Logf(t, "Output of Ycql command :- %s", output)
+		return "", nil
+	})
+}
+
+// Verify Master and Tserver conf exists or not. If exists then prints its Value.
+func testYugaByteConf(t *testing.T, terraformOptions *terraform.Options, sshUser string, yugabyteDir string, sshSession ssh.Host) {
+	masterConf := fmt.Sprintf("/home/%s/yugabyte-db/master/conf/server.conf", sshUser)
+	tserverConf := fmt.Sprintf("/home/%s/yugabyte-db/tserver/conf/server.conf", sshUser)
+
+	fdMasterConf, errFDMasterConf := os.Create(fmt.Sprintf("%s/master-server.conf", yugabyteDir))
+	if errFDMasterConf != nil {
+		t.Fatalf("Unable to create file descriptor for Master Conf%s", errFDMasterConf)
+	}
+
+	fdTserverConf, errFDTserverConf := os.Create(fmt.Sprintf("%s/tserver-server.conf", yugabyteDir))
+	if errFDTserverConf != nil {
+		t.Fatalf("Unable to create file descriptor for Tserver Conf%s", errFDTserverConf)
+	}
+
+	if getMasterFileError := ssh.ScpFileFromE(t, sshSession, masterConf, fdMasterConf, false); getMasterFileError != nil {
+		t.Fatalf("We got error while getting file from server :- Master: %s", getMasterFileError)
+	}
+
+	if getTserverFileError := ssh.ScpFileFromE(t, sshSession, tserverConf, fdTserverConf, false); getTserverFileError != nil {
+		t.Fatalf("We got error while getting file from server :- Tserver: %s", getTserverFileError)
+	}
+
+	outputMasterConf := ssh.CheckSshCommand(t, sshSession, fmt.Sprintf("cat /home/%s/yugabyte-db/master/conf/server.conf", sshUser))
+	outputTserverConf := ssh.CheckSshCommand(t, sshSession, fmt.Sprintf("cat /home/%s/yugabyte-db/tserver/conf/server.conf", sshUser))
+	logger.Logf(t, "Master Conf:\n%s", outputMasterConf)
+	logger.Logf(t, "Tserver Conf:\n%s", outputTserverConf)
+}
+
+// Check Tserver process is running or not
+func testYugaByteProcess(t *testing.T, terraformOptions *terraform.Options, sshSession ssh.Host) {
+	// Enhancement - Verify Master Process status
+	// checkMasterProcess := "pgrep --list-name 'yb-master'"
```

Comment:
maybe comment that the challenge here is because the master process only exists on a subset of the nodes.

Reply:
Yeah and terraform itself doesn't have the idea about which node will behave as a master in future. Okay, I'll add the comment for the same.